### PR TITLE
TVOS-178: Handling collection requests

### PR DESCRIPTION
**Pull request by huebnerob**
# Ticket(s)

https://vimean.atlassian.net/browse/TVOS-178

# Ticket Summary

Whenever we get a list of API resources, like videos or users, the response is paginated. There's a specific JSON response structure that handles returning not just the resources themselves, but also relevant metadata like the total items count and next/previous page links. We'd like to figure out the best way to wrap this interface with VimeoNetworking, and implement it.

# Implementation Summary

- `Request` now has `page` and `itemsPerPage` properties, which are respected when making the request with a client
- `Response` now has a multitude of pagination metadata properties, including total items count, and associated pages requests 

# How to test

Check out the sample project, I added a use of a next page request.  Also try making arbitrary requests like `Request(page: 3, itemsPerPage:83)`

### CODE REVIEW

File: VimeoNetworking/VimeoNetworking/Request.swift [hunk 1]
```diff
@@ -125,17 +156,42 @@ public struct Request<ModelType: MappableResponse>
...
     }
+    
+    // MARK: Copying requests
+    
+    internal func associatedPageRequest(newPath newPath: String) -> Request<ModelType>
+    {
+        let (path, query) = newPath.splitLinkString()
+        
+        let queryParameters = query?.parametersFromQueryString()
+        let page = Int((queryParameters?[self.PageKey]) ?? "")
```
Comment (jasonhawkins) [inline VimeoNetworking/VimeoNetworking/Request.swift]: Should these default values be set?

**Comment by huebnerob** [comment]
reassigning to @jasonhawkins

**Comment by huebnerob** [comment]
@alfiehanssen or @jasonhawkins can either one of y'all review this? 🙏

**Comment by jasonhawkins** [comment]
I took a look at this guy last week and it looks good beside the merge conflict. I was hesitant to give it a thumbs up because I'm not super familiar with the networking code or how this might affect it. I asked Alfie if he could take a look when he had some time, as a sanity check.

**Comment by jasonhawkins** [comment]
I went through this again and I'm good to give my 👍 once the merge conflicts are resolved. 

🎀

**Pull request merged**